sdk/storage/azure-storage-common/src/storage_retry_policy.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "azure/storage/common/internal/storage_retry_policy.hpp"
+
+namespace Azure { namespace Storage { namespace _internal {
+
+  bool StorageRetryPolicy::ShouldRetryOnResponse(
+      Core::Http::RawResponse const& response,
+      Core::Http::Policies::RetryOptions const& retryOptions,
+      int32_t attempt,
+      std::chrono::milliseconds& retryAfter,
+      double jitterFactor) const
+  {
+    bool ret = RetryPolicy::ShouldRetryOnResponse(
+        response, retryOptions, attempt, retryAfter, jitterFactor);
+    if (ret)
+    {
+      return true;
+    }
+
+    if (attempt > m_options.MaxRetries)
+    {
+      return false;
+    }
+
+    if (static_cast<std::underlying_type_t<Core::Http::HttpStatusCode>>(response.GetStatusCode())
+        >= 400)
+    {
+      const auto& headers = response.GetHeaders();
+      auto ite = headers.find("x-ms-copy-source-status-code");
```

Comment:
Add logging.

Reply:
`ShouldRetryOnTransportFailure` only evaluates a response should be retried or not. We still re-use retry logic in base class, and log is printed there.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/blob/1d50b1283ee5858e2b5a001c9981194566405666/sdk/core/azure-core/src/http/retry_policy.cpp#L165

Reply:
You're missing logging from similar to the following from the base`ShouldRetryOnResponse`:
https://github.com/Azure/azure-sdk-for-cpp/blob/1d50b1283ee5858e2b5a001c9981194566405666/sdk/core/azure-core/src/http/retry_policy.cpp#L228-L247

For example, its good to log that we are retrying because of the particular`innerStatusCode` found in the `x-ms-copy-source-status-code` header, before returning true on line 37.